src/controllers/krate/publish.rs

```diff
@@ -110,6 +91,51 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
         app.rate_limiter
             .check_rate_limit(user.id, rate_limit_action, conn)?;
 
+        let content_length = tarball_bytes.len() as u64;
+
+        let maximums = Maximums::new(
+            existing_crate.as_ref().and_then(|c| c.max_upload_size),
+            app.config.max_upload_size,
+            app.config.max_unpack_size,
+        );
+
+        if content_length > maximums.max_upload_size {
+            return Err(cargo_err(&format_args!(
+                "max upload size is: {}",
+                maximums.max_upload_size
+            )));
+        }
+
+        let pkg_name = format!("{}-{}", &*metadata.name, &*metadata.vers);
+        let tarball_info = process_tarball(&pkg_name, &*tarball_bytes, maximums.max_unpack_size)?;
+
+        // `unwrap()` is safe here since `process_tarball()` validates that
+        // we only accept manifests with a `package` section and without
+        // inheritance.
+        let package = tarball_info.manifest.package.unwrap();
+
+        let description = package.description.map(|it| it.as_local().unwrap());
+        let license = package.license.map(|it| it.as_local().unwrap());
+        let license_file = package.license_file.map(|it| it.as_local().unwrap());
+
+        // Make sure required fields are provided
+        fn empty(s: Option<&String>) -> bool {
+            s.map_or(true, String::is_empty)
+        }
+
+        // It can have up to three elements per below conditions.
+        let mut missing = Vec::with_capacity(3);
```

Comment:
The old code had the same behaviour, but aren't there two possible elements, not three?

Reply:
yeah, I noticed that too :D

the code will soon be replaced anyway, so I didn't feel like adjusting it :)

Reply:
Fine by me. We can probably afford the extra 16 bytes. :smiley: